### PR TITLE
[Step2] 문자열 덧셈 계산기 리뷰 요청 드립니다.

### PR DESCRIPTION
**Pull request by chae-yh**
### Description
* 스텝 2 문자열 덧셈 계산기 입니다. 문자열로 입력을 받은 값에 대해 구분자 기준으로 잘라 합을 리턴합니다.

### Commits
* docs : update docs for step 1 and list up to-do list for step 2
* feat : add new test calss in which it tests, when it gets null or empty string, whether it returns 0
* docs : update docs about test cases that are about response for null or empty string
* docs : update docs where when it gets one number(and no comma or colon), then it returns the number
* feat : add a function in a class where when it gets one number, then it returns the number. And add test class
* feat : add a function where it splits input by comma or colon and returns sum of them
* feat : add number class that conatins validation login in it. and number class can not accept negative or string
* docs : update to-do list that related to validation logic
* feat : add validation logic in number class where it can not exceed specific number
* feat : create 1st collection for numbers that has its own validation logic and add test code
* feat : create delimeter class that contains its validation logic and add test code
* refactor : refactor source in order not to use getter method
* feat : add input wrapping class and move some logic to it
* fix : rename variables and modify access modifier

### CODE REVIEW

**Review by ssosso — CHANGES_REQUESTED** [review]
영훈님, 안녕하세요:)
피드백 남겨드리오니 확인부탁드립니다~
그리고 요구사항에 몇 가지 제약들을 더 추가하여 여러 클래스들로 구현해주셨는데, StringAddCalculatorTest에 있는 테스트를 각 클래스의 테스트로 변경해보면 어떨까요~?
굳이 StringAddCalculator를 통하지 않아도 각각 테스트할 수 있습니다:)

File: src/main/java/StringAddCalculator/Input.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		return split(input, new Delimiter());
+	}
+
+	private String[] split(String input, Delimiter delimiter) {
+		return delimiter.splitInput(input);
+	}
+
+	private static boolean validateInput(String input) {
+		if (input == null || input.isEmpty()) {
+			return true;
```
[the remark this answers is not in the document]
Comment (chae-yh) [reply src/main/java/StringAddCalculator/Input.java:38]: 말씀하신대로, 그냥 validation 으로는 의미가 명확하지 않은 듯 싶습니다.
true / false 를 구분하는 형식의 isNullOrEmpty 정도의 메소드로 변경하겠습니다.  
확인 감사합니다 : )

**Comment by chae-yh** [comment]
피드백 내용 반영하였고, 피드백 내용 반영 중 수정 사항 몇가지 추가하였습니다. : )
수정 사항은 하기와 같습니다.
1. 피드백 내용 반영 : validationInput 메소드명 변경(validationInput -> isNullOrEmpty)
2. 매직 넘버 상수로 변경
3. 각 클래스 별 테스트 코드 추가(validation 및 로직 테스트)
- 기존 StringAddCalculatorTest에 있던 코드는 전체적인 코드 수행이 정상적인지 확인하기 위하여 그대로 두었습니다.

StringAddCalculatorTest 를 먼저 작성하고 TDD를 통해 요구사항에 해당 하는 코드를 작성한 후 각 클래스를 분리하는 리펙토링이 진행되니 각 클래스별 테스트 코드가 미흡했습니다. 이번에 다시 테스트 코드 분리 작업을 하며 각 클래스에 대해 조금 더 고민하게 되어 좋았던 것 같습니다 : )

**Review by ssosso — APPROVED** [review]
영훈님~ 재요청주신 코드 확인하였습니다:)
작성해주신 내용도 읽어보았는데, 제가 의도한 바를 잘 이해해주신거 같아 기쁩니다!
다음 단계진행하실 수 있도록 이만 merge하도록 하겠습니다, 고생 많으셨습니다~
